src/backend/base/langflow/services/job_queue/service.py

```diff
@@ -0,0 +1,248 @@
+            msg = f"No queue found for job_id {job_id}"
+            logger.error(msg)
+            raise ValueError(msg)
+
+        if self._closed:
+            msg = "Queue service is closed"
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        main_queue, event_manager, existing_task = self._queues[job_id]
+
+        if existing_task and not existing_task.done():
+            logger.debug(f"Existing task for job_id {job_id} detected; cancelling it.")
+            existing_task.cancel()
+
+        # Initiate the new asynchronous task.
+        task = asyncio.create_task(task_coro)
+        self._queues[job_id] = (main_queue, event_manager, task)
+        logger.debug(f"New task started for job_id {job_id}")
+
+    def get_queue_data(self, job_id: str) -> tuple[asyncio.Queue, EventManager, asyncio.Task | None]:
+        """Retrieve the complete data structure associated with a job's queue.
+
+        Args:
+            job_id (str): Unique identifier for the job.
+
+        Returns:
+            tuple[asyncio.Queue, EventManager, asyncio.Task | None]:
+                A tuple containing the job's main queue, its linked event manager, and the associated task (if any).
+        """
+        if job_id not in self._queues:
+            msg = f"No queue found for job_id {job_id}"
+            logger.error(msg)
+            raise ValueError(msg)
+
+        if self._closed:
+            msg = "Queue service is closed"
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        return self._queues[job_id]
+
+    async def cleanup_job(self, job_id: str) -> None:
+        """Clean up and release resources for a specific job.
+
+        The cleanup process includes:
+          1. Verifying if the job's queue is registered.
+          2. Cancelling the running task (if active) and awaiting its termination.
+          3. Clearing all items from the job's queue.
+          4. Removing the job's entry from the internal registry.
+
+        Args:
+            job_id (str): Unique identifier for the job to be cleaned up.
+        """
+        if job_id not in self._queues:
+            logger.debug(f"No queue found for job_id {job_id} during cleanup.")
+            return
+
+        logger.info(f"Commencing cleanup for job_id {job_id}")
+        main_queue, event_manager, task = self._queues[job_id]
+
+        # Cancel the associated task if it is still running.
+        if task and not task.done():
+            logger.debug(f"Cancelling active task for job_id {job_id}")
+            task.cancel()
+            await asyncio.wait([task])
+            if not task.cancelled():
+                exc = task.exception()
+                if exc is not None:
+                    raise exc
+            logger.debug(f"Task cancellation complete for job_id {job_id}")
+
+        # If there is no active task or the task has completed, proceed with clearing the queue.
+        if task is None or task.done():
```

Comment:
That `if` could be removed ?

Reply:
Makes sense. You say that because we can call the clean up everytime, right?